Something/src/main/java/org/example/Menu.java

```diff
@@ -0,0 +1,46 @@
+package org.example;
+
+import java.util.*;
+
+public class Menu {
+   /* String placeholder;*/
+    Map<String, MenuDetails> menuMap = new LinkedHashMap<String, MenuDetails>();
+    List<String> isTrue = new ArrayList<>();
+
+    public Map<String, MenuDetails> getMenu() {
+        return menuMap;
+    }
+
+    public void setMenu(Map<String, MenuDetails> menu) {
+        this.menuMap = menu;
+    }
+
+    public List<String> getVisibleItems(){
+        List<String> visibleItems = new ArrayList<>();
+        for(MenuDetails i : menuMap.values()){
+            if(i.getVisible()){
```

Comment:
Using a single character for a variable is not 'bad' but when you have a loop that spans a few lines and you are looking for 'i' then you can sometimes miss it.

In this case, maybe variable could be called menuDetail because thats what it is

Reply:
Is menuDetail not already used for something else? "i" was chosen as it seems to be the standard when using for loops especially since you can go along the alphabet with more for loops within the same method and you could reuse the same var names in for loops in other methods.